### PR TITLE
checkpoint_abort script does not exist on the mom

### PR DESCRIPTION
**Pull request by birbalsain**
#### Describe Bug or Feature
checkpoint_abort script does not exist on the remote mom.

#### Describe Your Change
create the temp file with the mom hostname.

#### Attach Test and Valgrind Logs/Output

[checkpoint_abort_file.txt](https://github.com/PBSPro/pbspro/files/4025248/checkpoint_abort_file.txt)

### CODE REVIEW

File: test/tests/functional/pbs_checkpoint.py [hunk 1]
```diff
@@ -144,7 +148,7 @@ def test_checkpoint_abort_with_preempt(self):
         self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
 
         j1 = Job(TEST_USER, self.attrs)
-        j1.set_sleep_time(20)
+        j1.set_sleep_time(80)
```
Comment (arungrover) [inline test/tests/functional/pbs_checkpoint.py]: why did you increase the sleep time of the job?

File: test/tests/functional/pbs_release_limited_res_suspend.py [hunk 2]
```diff
@@ -989,10 +989,13 @@ def test_preempt_checkpoint_release_all(self):
                 exit 0
                 """
         pbs_home = self.server.pbs_conf['PBS_HOME']
```
Comment (arungrover) [inline test/tests/functional/pbs_release_limited_res_suspend.py]: shouldn't this be self.mompbs_conf['PBS_HOME'] since you are making a script on mom?

File: test/tests/functional/pbs_checkpoint.py (hunk 1 above)
Comment (birbalsain) [reply test/tests/functional/pbs_checkpoint.py]: Because we need, both job must be available. but first job get finished before second job is started.

File: test/tests/functional/pbs_release_limited_res_suspend.py (hunk 2 above)
Comment (birbalsain) [reply test/tests/functional/pbs_release_limited_res_suspend.py]: we are just checking pbs home directory.

File: test/tests/functional/pbs_checkpoint.py (hunk 1 above)
Comment (arungrover) [reply test/tests/functional/pbs_checkpoint.py]: Why is it that the first job finishes before second starts? I don't see any place in the test which would cause a delay unless there is some problem with the system itself.
In any case, If you really want to increase the sleep time, I'd say just don't set it. By default, every job sleeps for 100 seconds.

File: test/tests/functional/pbs_release_limited_res_suspend.py (hunk 2 above)
Comment (arungrover) [reply test/tests/functional/pbs_release_limited_res_suspend.py]: No, you are passing this PBS_HOME read from server's pbs.conf to create a file on mom.  Ideally you should be using PBS_HOME read from mom's pbs.conf file.

File: test/tests/functional/pbs_checkpoint.py (hunk 1 above)
Comment (bhroam) [reply test/tests/functional/pbs_checkpoint.py]: Or if you don't need the to wait for the jobs to end, increase it longer than 100 seconds.

File: test/tests/functional/pbs_checkpoint.py [hunk 3]
```diff
@@ -55,9 +55,12 @@ def setUp(self):
...
 """
-        self.abort_file = self.du.create_temp_file(body=abort_script)
-        self.du.chmod(path=self.abort_file, mode=0o755)
-        self.du.chown(path=self.abort_file, uid=0, gid=0, runas=ROOT_USER)
+        self.abort_file = self.du.create_temp_file(hostname=self.mom.hostname,
+                                                   body=abort_script)
+        self.du.chmod(hostname=self.mom.hostname,
+                      path=self.abort_file, mode=0o755)
+        self.du.chown(hostname=self.mom.hostname, path=self.abort_file,
+                      uid=0, gid=0, runas=ROOT_USER)
```
Comment (bhroam) [inline test/tests/functional/pbs_checkpoint.py]: Since this is a common mistake, maybe we should create a function as part of the mom similar to the function that creates a mom dynamic resource.  This one will create the checkpoint script, chmod/chown it, and set preempt_prio on the server.  This way people can always use this instead of this mistake popping up from time to time.

File: test/tests/functional/pbs_checkpoint.py (hunk 1 above)
Comment (birbalsain) [reply test/tests/functional/pbs_checkpoint.py]: > Why is it that the first job finishes before second starts? I don't see any place in the test which would cause a delay unless there is some problem with the system itself.
> In any case, If you really want to increase the sleep time, I'd say just don't set it. By default, every job sleeps for 100 seconds.

Some platforms take long time to do job process.

Comment (arungrover) [reply test/tests/functional/pbs_checkpoint.py]: Hmmm, that seems odd that PBS could not preempt the job within 20 seconds of running one job when the test does not even have any sleep. In any case, just don't set and it will be set to 100 seconds by default.

File: test/tests/functional/pbs_checkpoint.py (hunk 3 above)
Comment (birbalsain) [reply test/tests/functional/pbs_checkpoint.py]: @bhroam check the new commit.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 4]
```diff
@@ -13452,6 +13453,21 @@ def create_vnode_def(self, name, attrs={}, numnodes=1, sharednode=True,
...
+        """
+        Add checkpoint script in the mom config.
+        returns: a temp file for checkpoint script
+        """
+        chk_file = self.du.create_temp_file(hostname=self.hostname, body=body,
+                                            dirname=dirname)
+        self.du.chmod(hostname=self.hostname, path=chk_file, mode=mode)
+        self.du.chown(hostname=self.hostname, path=chk_file, uid=uid,
+                      gid=gid, sudo=sudo, runas=runas)
+        c = {'$action': 'checkpoint_abort 30 !' + chk_file + ' %sid'}
```
Comment (bhroam) [inline test/fw/ptl/lib/pbs_testlib.py]: Add 30 as a default argument.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 5]
```diff
@@ -13452,6 +13453,21 @@ def create_vnode_def(self, name, attrs={}, numnodes=1, sharednode=True,
         del attribs
         return "\n".join(vdef)
 
+    def add_checkpoint_script(self, dirname=None, body=None, mode=None,
```
Comment (bhroam) [inline test/fw/ptl/lib/pbs_testlib.py]: There are more than one checkpoint $action scripts in the mom config.  Call this function add_checkpoint_abort_script()

File: test/fw/ptl/lib/pbs_testlib.py [hunk 6]
```diff
@@ -13452,6 +13453,21 @@ def create_vnode_def(self, name, attrs={}, numnodes=1, sharednode=True,
...
+                              runas=None, sudo=False, uid=None, gid=None):
```
Comment (bhroam) [inline test/fw/ptl/lib/pbs_testlib.py]: I don't think you need runas, sudo, mode, uid, or gid.  The mom will complain if the script isn't secure.  I think we should force these to be root with secure permissions.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 7]
```diff
@@ -68,16 +68,16 @@
 try:
     import psycopg2
     PSYCOPG = True
-except:
+except BaseException:
```
Comment (bhroam) [inline test/fw/ptl/lib/pbs_testlib.py]: Why the change from the blanket excepts to BaseException?  The reason you don't do blanket excepts is because it catches things like KeyboardInterrupt.  BaseException includes that.  This isn't the right PR to do this change.  When we get rid of the blanket excepts in pbs_testlib.py, we should go through and actually look what the try/except block is expects and catch that.

File: test/tests/functional/pbs_release_limited_res_suspend.py [hunk 8]
```diff
@@ -988,17 +988,14 @@ def test_preempt_checkpoint_release_all(self):
...
-                                                 body=chk_script,
-                                                 dirname=pbs_home)
-        self.du.chmod(hostname=self.mom.hostname,
-                      path=self.chk_file, mode=0o755)
-        self.du.chown(hostname=self.mom.hostname,
-                      path=self.chk_file, uid=0, gid=0, sudo=True)
-        c = {'$action': 'checkpoint_abort 30 !' + self.chk_file + ' %sid'}
-        self.mom.add_config(c)
-
+        pbs_home = self.mom.pbs_conf['PBS_HOME']
```
Comment (birbalsain) [inline test/tests/functional/pbs_release_limited_res_suspend.py]: @arungrover updated pbs_home.

File: test/tests/functional/pbs_checkpoint.py [hunk 9]
```diff
@@ -148,7 +142,6 @@ def test_checkpoint_abort_with_preempt(self):
         self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
 
         j1 = Job(TEST_USER, self.attrs)
-        j1.set_sleep_time(80)
```
Comment (birbalsain) [inline test/tests/functional/pbs_checkpoint.py]: @arungrover , remove the set sleep time

File: test/fw/ptl/lib/pbs_testlib.py (hunk 7 above)
Comment (birbalsain) [reply test/fw/ptl/lib/pbs_testlib.py]: @bhroam , removed the changes.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 10]
```diff
@@ -13453,18 +13453,17 @@ def create_vnode_def(self, name, attrs={}, numnodes=1, sharednode=True,
         del attribs
         return "\n".join(vdef)
 
-    def add_checkpoint_script(self, dirname=None, body=None, mode=None,
-                              runas=None, sudo=False, uid=None, gid=None):
+    def add_checkpoint_abort_script(self, dirname=None, body=None,
+                                    mode=None, abort_time=30):
```
Comment (bhroam) [inline test/fw/ptl/lib/pbs_testlib.py]: Passing mode=None causes chmod() to fail.  We know what will work for a checkpoint_abort script.  For security reasons, I'd suggest we use 0o700 instead of 0o755.

Comment (birbalsain) [reply test/fw/ptl/lib/pbs_testlib.py]: I think, it is good to remove  mode as a argument. we can direct use mode=0o700 in chmod().

File: test/tests/functional/pbs_release_limited_res_suspend.py (hunk 8 above)
Comment (birbalsain) [reply test/tests/functional/pbs_release_limited_res_suspend.py]: @arungrover  I addressed your requested changes
